apollo-federation/src/sources/connect/json_selection/methods.rs

```diff
@@ -0,0 +1,2178 @@
+        }
+        JSON::String(s) => {
+            let size = s.as_str().len() as i64;
+            tail.apply_to_path(&JSON::Number(size.into()), vars, input_path, errors)
+        }
+        // Though we can't ask for ->first or ->last or ->at(n) on an object, we
+        // can safely return how many properties the object has for ->size.
+        JSON::Object(map) => {
+            let size = map.len() as i64;
+            tail.apply_to_path(&JSON::Number(size.into()), vars, input_path, errors)
+        }
+        _ => {
+            errors.insert(ApplyToError::new(
+                format!(
+                    "Method ->{} requires an array, string, or object input, not {}",
+                    method_name,
+                    json_type_name(data),
+                ),
+                input_path.to_vec(),
+            ));
+            None
+        }
+    }
+}
+
+// Returns a list of [{ key, value }, ...] objects for each key-value pair in
+// the object. Returning a list of [[ key, value ], ...] pairs might also seem
+// like an option, but GraphQL doesn't handle heterogeneous lists (or tuples) as
+// well as it handles objects with named properties like { key, value }.
+fn entries_method(
+    method_name: &str,
+    method_args: &Option<MethodArgs>,
+    data: &JSON,
+    vars: &VarsWithPathsMap,
+    input_path: &InputPath<JSON>,
+    tail: &PathList,
+    errors: &mut IndexSet<ApplyToError>,
+) -> Option<JSON> {
+    if let Some(MethodArgs(_)) = method_args {
+        errors.insert(ApplyToError::new(
+            format!("Method ->{} does not take any arguments", method_name),
+            input_path.to_vec(),
+        ));
+        return None;
+    }
+
+    match data {
+        JSON::Object(map) => {
+            let entries = map
+                .iter()
+                .map(|(key, value)| {
+                    let mut key_value_pair = JSONMap::new();
+                    key_value_pair.insert(ByteString::from("key"), JSON::String(key.clone()));
+                    key_value_pair.insert(ByteString::from("value"), value.clone());
+                    JSON::Object(key_value_pair)
+                })
+                .collect();
+            tail.apply_to_path(&JSON::Array(entries), vars, input_path, errors)
+        }
+        _ => {
+            errors.insert(ApplyToError::new(
+                format!(
+                    "Method ->{} requires an object input, not {}",
+                    method_name,
+                    json_type_name(data),
+                ),
+                input_path.to_vec(),
+            ));
+            None
+        }
+    }
+}
+
+// The future module contains methods that are not yet exposed for use in
+// JSONSelection strings in connector schemas, but are implemented and tested.
+// Before any of these methods graduates to public status, it should be reviewed
+// carefully and then moved to the top level of the methods.rs file (out of the
+// future:: namespace).
+mod future {
```

Comment:
Can you move this to a sibling file so that this file gets shorter? Reminder that I'm afraid of big files because of the painful Rust 1.76 compilation bug 😓. I _think_ if you put it in a file called `methods/future.rs` then the rest of the code should be almost identical.

Obviously that's a temporary solution... so if you have another idea at how we might split up the implementation longer-term, that could work too. Maybe grouping based on the general vibe of the method (logical, arithmetic, etc.)?

Reply:
Done: https://github.com/apollographql/router/pull/5762/commits/60b877782389196f60ae18247ec0809da53aeb11